Regular Expressions/RegExp (Basic).sublime-syntax

```diff
@@ -0,0 +1,280 @@
+      scope: constant.other.assertion.regexp
+      set: [group-body-extended, maybe-unexpected-quantifiers]
+    # Deactivates 'x' mode
+    - match: (\?[ims]*-[ims]*x[imxs]*)(\))
+      captures:
+        1: storage.modifier.mode.regexp
+        2: keyword.control.group.regexp
+      set: [base-group, maybe-unexpected-quantifiers]
+    # Groups without 'x' mode
+    - match: '\?[ims]*-[ims]*x[imxs]*:'
+      scope: storage.modifier.mode.regexp
+      set: [group-body, maybe-unexpected-quantifiers]
+    # Other modifiers
+    - match: '\?(?:[ixms]*-)?[ims]+:'
+      scope: storage.modifier.mode.regexp
+      set: [group-body-extended, maybe-unexpected-quantifiers]
+    - include: group-start-common
+    - match: ''
+      set: [group-body-extended, maybe-unexpected-quantifiers]
+
+  group-start-common:
+    # Other modifiers (must come after other modifier matches)
+    - match: (\?(?:[ixms]*-)?[ixms]+)(\))
+      captures:
+        1: storage.modifier.mode.regexp
+        2: keyword.control.group.regexp
+      pop: 1
+
+  group-body:
+    - meta_content_scope: meta.group.regexp
+    - match: \)
+      scope: meta.group.regexp keyword.control.group.regexp
+      pop: 1
+    - include: base-literal
+
+  group-body-extended:
+    - meta_content_scope: meta.group.extended.regexp
+    - match: \)
+      scope: meta.group.extended.regexp keyword.control.group.regexp
+      pop: 1
+    - include: base-literal-extended
+
+  charsets:
+    - match: (\[\^?)\]?
+      captures:
+        1: keyword.control.set.regexp
+      push: charset-body
+
+  charset-body:
+    - meta_scope: meta.set.regexp
+    - match: \]
+      scope: keyword.control.set.regexp
+      pop: 1
+    - include: charset-ranges
+    - include: character-classes
+    - include: escaped-chars
+
+  charset-ranges:
+    - match: (?=({{known_char_escape}}|{{char_escape}}|(?!\\-)[^\]])-({{known_char_escape}}|{{char_escape}}|[^\]]))
+      push: [charset-range-end, charset-range-begin]
+
+  charset-range-begin:
+    - match: '-'
+      scope: punctuation.separator.sequence.regexp-common
+      pop: 1
+    - include: escaped-chars
+
+  charset-range-end:
+    - meta_scope: constant.other.range.regexp
+    - include: escaped-char
+    - match: .
+      pop: 1
+
+  character-classes:
+    - match: '{{char_class}}'
+      scope: keyword.control.character-class.regexp
+
+  escaped-chars:
+    - match: '{{known_char_escape}}'
+      scope: constant.character.escape.regexp
+    - match: '{{invalid_char_escape}}'
+      scope: invalid.illegal.character.escape.regexp
+    - match: '{{char_escape}}'
+      scope: constant.character.escape.regexp
+
+  escaped-char:
+    - match: '{{known_char_escape}}'
+      scope: constant.character.escape.regexp
+      pop: 1
+    - match: '{{invalid_char_escape}}'
+      scope: invalid.illegal.character.escape.regexp
+      pop: 1
+    - match: '{{char_escape}}'
+      scope: constant.character.escape.regexp
+      pop: 1
+
+  anchors:
+    - match: \\[bBAZzG]|[\^$]
```

Comment:
Do we want to create a variable for this pattern, too? Might help inherited syntaxes to override just the pattern without the need to address the whole context.

Reply:
I'm of two minds about this - having a variable that is only used in one place doesn't add much value to the base syntax. And overriding it in the extended syntax makes it less obvious what was changed/added - one would have to compare it to the variable in the base syntax. So generally I prefer to append/prepend rules to the context, especially as then one can change the scope suffix to make it clear which syntax it came from.

But making things easy to modify with variables simplifies extending - no need to understand the contexts, or anything in the base syntax.

Reply:
That's the general issue with variables, yes. Especially if they contain capture groups. The context is small enough to be easily replaced by an inherited syntax. Maybe just appended/prepended to add more stuff.

I have no strong opinion about it, while sharing your concerns. It was just an idea due to all the other variables already existing.